### ISSUE TITLE
Hope to support Android

### ISSUE THREAD

**Issue by mzdk100** [compio-rs/winio#49]
There are also many UI libraries in Rust that support Android, but none of them use native Android widgets, so I hope Winio can support them.
The reason why I want to find a native UI library is because many UI libraries lack accessibility support, and native UI does not have this problem.

**Comment by mzdk100** [comment]
Will they consider it in the future? If necessary, I can also participate.
I am not yet familiar with the framework of the entire project, so it may be difficult for me to implement it myself.

**Comment by Berrysoft** [comment]
I don't think it will be considered in the near future. If you want to participate, I suggest you first trying to write some JNI with Rust targeting Android, and then trying to write a basic native Android activity in Rust. If you succeed, we may have enough confidence to consider supporting Android in winio.

**Comment by mzdk100** [comment]
Basic native activities are now available, and you can find them [here](https://github.com/mzdk100/droid-wrap/blob/master/example/src/activity.rs)

**Comment by Berrysoft** [comment]
OK. It seems a usable wrapper. Then the problem will be, how do we express the app lifecycle? We need to consider the design.

**Comment by mzdk100** [comment]
I am not very familiar with the framework of Winio yet. If I want to create a winio-ui-android crate, I don't know what necessary interfaces must be implemented. Currently, there are few documents on winio.
Or can Winio design a plugin mechanism so that there is no need to modify too much framework code, only platform specific plugins need to be imported.

**Comment by Berrysoft** [comment]
The needed interfaces are constrained by the `winio` crate. It just pulls all needed types and method in. There's no plugin mechanism because it's hard and messive to do so.

It might be an easy start to edit `winio/src/lib.rs` first to add your own backend, and then `winio-handle` to add your own window & widget handle types. The `winio-ui-*` crates only provide two sets: the runtime and the widgets (including windows). The impl of the runtimes are alike, and you might want to take the app-kit one or gtk one as references. Considering Android doesn't expose enough APIs to poll the runtime, you might also need to do the work like winui backend, which polls the runtime in another thread.

Sorry but I don't have time to write a full document.
